plugin/gcp/instance.py

```diff
@@ -0,0 +1,239 @@
+
+    @blocking(True)
+    def remove_tags(self, tags):
+        """
+        Remove GCP instance tags.
+        Zone operation.
+
+        :return: REST response with operation responsible for the instance
+        tag removal process and its status
+        """
+        # each tag should be RFC1035 compliant
+        self.logger.info('Remove tags')
+        self.tags = [tag for tag in self.tags if tag not in tags]
+        fingerprint = self.get()["tags"]["fingerprint"]
+        return self.compute.instances().setTags(
+            project=self.project,
+            zone=self.zone,
+            instance=self.name,
+            body={"items": self.tags, "fingerprint": fingerprint}).execute()
+
+    def get(self):
+        """
+        Get GCP instance details.
+        """
+        self.logger.info("Get instance details")
+        return self.compute.instances().get(
+            instance=self.name,
+            project=self.project,
+            zone=self.zone).execute()
+
+    @blocking(True)
+    def add_access_config(self):
+        """
+        Set GCP instance external IP.
+        Zone operation.
+
+        :return: REST response with operation responsible for the instance
+        external IP setting process and its status
+        """
+        body = {"kind": "compute#accessConfig",
+                "name": self.ACCESS_CONFIG,
+                "type": self.ACCESS_CONFIG_TYPE}
+        return self.compute.instances().addAccessConfig(
+            project=self.project,
+            instance=self.name,
+            zone=self.zone,
+            networkInterface=self.NETWORK_INTERFACE,
+            body=body).execute()
+
+    def delete_access_config(self):
+        """
+        Set GCP instance tags.
+        Zone operation.
+
+        :return: REST response with operation responsible for the instance
+        tag setting process and its status
+        """
+        return self.compute.instances().deleteAccessConfig(
+            project=self.project,
+            instance=self.name,
+            zone=self.zone,
+            accessConfig=self.ACCESS_CONFIG,
+            networkInterface=self.NETWORK_INTERFACE).execute()
+
+    def list(self):
+        """
+        List GCP instances.
+        Zone operation.
+
+        :return: REST response with a list of instances and its details
+        """
+        self.logger.info("List instances")
+        return self.compute.instances().list(
+            project=self.project,
+            zone=self.zone).execute()
+
+    def wait_for_operation(self, operation, global_operation=False):
+        super(Instance, self).wait_for_operation(operation, global_operation)
+
+    def to_dict(self):
+        body = {
+            'name': self.name,
+            'machineType': 'zones/{0}/machineTypes/{1}'.format(
+                self.zone,
+                self.machine_type),
+            'disks': [
+                {
+                    'boot': True,
```

Comment:
They will, I'm preparing the next branch 'add disk'. It will allow disk creation/deletion and later snapshots management.